src/npc.cpp

```diff
@@ -2459,3 +2456,55 @@ std::string npc::get_companion_mission() const
 {
     return companion_mission;
 }
+
+attitude_group get_attitude_group( npc_attitude att )
+{
+    switch( att ) {
+        case NPCATT_MUG:
+        case NPCATT_WAIT_FOR_LEAVE:
+        case NPCATT_KILL:
+            return attitude_group::hostile;
+        case NPCATT_FLEE:
+            return attitude_group::fearful;
+        case NPCATT_FOLLOW:
+        case NPCATT_LEAD:
+            return attitude_group::friendly;
+        default:
+            break;
+    }
+    return attitude_group::neutral;
+}
+
+npc_attitude npc::get_attitude() const
+{
+    return attitude;
+}
+
+void npc::set_attitude( npc_attitude new_attitude )
+{
+    if( new_attitude == attitude ) {
+        return;
+    }
+    add_msg( m_debug, "%s changes attitude from %s to %s",
+             npc_attitude_name( attitude ).c_str(), npc_attitude_name( new_attitude ).c_str() );
+    attitude_group new_group = get_attitude_group( new_attitude );
+    attitude_group old_group = get_attitude_group( attitude );
+    if( new_group != old_group && !is_fake() && g->u.sees( *this ) ) {
+        switch( new_group ) {
+            case attitude_group::hostile:
+                add_msg( m_bad, _( "%s gets angry!" ), name.c_str() );
```

Comment:
There's `<npcname>` tag for this purpose. I suppose, it's more convenient for the translators since its self-explanatory. Are there particular reasons not to use it?